opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -130,6 +131,13 @@ def attributes(self) -> types.Attributes:
         pass
 
 
+class LinkDict(typing.TypedDict):
```

Comment:
`typing.TypedDict` was introduced [in 3.8](https://docs.python.org/3/library/typing.html?highlight=typeddict#typing.TypedDict). This is why tests are failing for `pypy` and 3.7.